numba/targets/arrayobj.py

```diff
@@ -1838,6 +1838,31 @@ def np_unique_impl(a):
     return np_unique_impl
 
 
+@overload(np.repeat)
+def np_repeat(a, repeats):
+    def np_repeat_impl(a, repeats):
+        a = a.ravel()
+        shape = a.shape
+        n = shape[0]
+        return_size = n*repeats
+        to_return = np.empty(return_size)
+        # if a.ndim > 1:
+        #     to_return = np.zeros((return_size, shape[1]))
+        # else:
+        #     to_return = np.zeros(return_size)
+        for i in range(n):
+            count = 1
+            ind = i*repeats
+            to_return[ind] = a[i]
+            while count < repeats:
```

Comment:
Also, I think this can be done using slice syntax in one line if desired?
```python
    for i in range(a.size):
        to_return[i * repeats : (i + 1) * repeats] = a.ravel()[i]
```

Reply:
So I tried the following on the original version and a loop optimized one and there doesn't seem to be any significant difference.

Here is the setup code:

```
In [1]: import numba as nb

In [2]: import numpy as np

In [3]: def re(a, r):
   ...:     return np.repeat(a, r)
   ...:

In [4]: nnr = nb.njit(re)

In [5]: a = np.arange(10000)
```

Here is the output of the original:

```
In [12]: %timeit nnr(a, 100)
6.38 µs ± 51.3 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

In [13]: %timeit nnr(a, 1000)
65 µs ± 3.27 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)

In [14]: %timeit nnr(a, 10000)
570 µs ± 3.6 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

In [15]: %timeit nnr(a, 100000)
11.1 ms ± 165 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)

In [16]: %timeit nnr(a, 1000000)
386 ms ± 13.8 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

And here for the optimized one:

```
In [12]: %timeit nnr(a, 100)
6.32 µs ± 62.5 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

In [13]: %timeit nnr(a, 1000)
60.4 µs ± 1.04 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)

In [14]: %timeit nnr(a, 10000)
571 µs ± 6.81 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

In [15]: %timeit nnr(a, 100000)
11 ms ± 183 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)

In [16]: %timeit nnr(a, 1000000)
360 ms ± 11.8 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

I also tried the slicing version and that seemed to yield slightly better results for smaller inputs:

```In [12]: %timeit nnr(a, 100)
1.59 µs ± 47 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [13]: %timeit nnr(a, 1000)
20.5 µs ± 392 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)

In [14]: %timeit nnr(a, 10000)
192 µs ± 12 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

In [15]: %timeit nnr(a, 100000)
9.81 ms ± 267 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)

In [16]: %timeit nnr(a, 1000000)
338 ms ± 14.1 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

I will add some more tests before I decide on a final version.